libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -185,14 +185,9 @@ class NavigationRoute internal constructor(
         ): List<NavigationRoute> {
             logI("NavigationRoute.createAsync is called", LOG_CATEGORY)
             return coroutineScope {
-                val deferredResponseParsing = async(ThreadController.DefaultDispatcher) {
-                    DirectionsResponse.fromJson(directionsResponseJson.asReader()).also {
-                        logD(
-                            "parsed directions response to java model for ${it.uuid()}",
-                            LOG_CATEGORY
-                        )
-                    }
-                }
+                val parsedResponse = async(ThreadController.DefaultDispatcher) {
+                    directionsResponseJson.toDirectionsResponse()
+                }.await()
```

Comment:
The important part of the fix: do not do 2 parsings in parallel: buffer position might end up messed up.